src/Pages/VideoListing/VideoListing.jsx

```diff
@@ -1,13 +1,24 @@
 import axios from "axios";
 import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
 import { Header } from "../../components/Header/Header";
+import { Sidebar } from "../../components/Sidebar/Sidebar";
+import { useAuth } from "../../Context/auth-context";
+import { useWatchLater } from "../../Context/watch-leter-context";
+import { addToWatchLater } from "../../Utility/addToWatchLater";
+import { deleteFromWatchLater } from "../../Utility/deleteFromWatchLater";
+import { Link } from "react-router-dom";
 import "../VideoListing/VideoListing.css";
 function VideoListing() {
   const [videos, setVideos] = useState([]);
+  const { watchLaterState, watchLaterDispatch } = useWatchLater();
+  const { watchedVideo } = watchLaterState;
+  const { auth } = useAuth();
+  const navigate = useNavigate();
+
   const getVideos = async () => {
     try {
       const { data } = await axios.get("/api/videos");
```

Comment:
you can add axios.defaults.baseUrl="/api/" one time and export then import and do as axios.get(/"videos") (Just a suggestion)